ls1intum/Artemis
# [Bugfix] - Don't count quiz submissions in practise mode into reachable score in course statistics

### Checklist

### Motivation and Context


Fixes https://github.com/ls1intum/Artemis/issues/2840. When Opening a quiz for practise, participations would count into the reachable score, which then surpasses the maximum score. Practise submissions should not be counted.

### Description

- Added a check in the course statistics
- Prior participations of quizzes are getting overwritten with intializationstate = INITIALIZED, which is changed to FINISHED

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a quiz and participate
4. Check that the quiz in the course statistics is displayed correctly
5. Open this quiz for practice (By going to course Management -> exercises -> Release for practice)
6. Submit a practise submission
7. Check that in the course statistics, the reachable score is correct and didnt change since step 4.
8. Create another quiz like mentioned above, but dont participate in the official quiz.
9. After the quiz has ended, release the quiz for practice and make a practice submission
10. The reachable score should be correct


### Screenshots


Participations in practise quizzes no longer count into reachable score:
![Screenshot 2021-02-08 at 22 35 41](https://user-images.githubusercontent.com/44401560/107284977-d05a9700-6a5e-11eb-9f04-7cae3c2d904a.png)
